rspirv/sr/context.rs

```diff
@@ -12,11 +12,94 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::{
+    collections::HashMap,
+    hash::BuildHasherDefault,
+    marker::PhantomData,
+};
+
+use fxhash::FxHasher;
 use spirv;
 
-use super::{Type, TypeToken, Constant, ConstantToken};
-use sr::constants::ConstantEnum;
-use sr::types::{StructMember, TypeEnum};
+use super::{
+    types::{Type, StructMember, TypeEnum},
+    constants::{Constant, ConstantEnum},
+    instructions::Instruction,
+    items::{Function, Variable},
+    structs,
+    LiftError, OperandError,
+};
+use crate::mr;
+
+type FastHashMap<K, V> = HashMap<K, V, BuildHasherDefault<FxHasher>>;
+type Index = u32;
+
+/// A structure holding some kind of a structured SPIRV data that can be
```

Comment:
Nit: SPIR-V is the formal spelling. Unless we are in code where we cannot use -, let's use the formal name when possible.

Reply:
... some kind of SPIR-V constructs that can ...